tasks/fr/fr_income.py

```diff
@@ -0,0 +1,197 @@
+
+
+class RawIncomeIrisData(CSV2TempTableTask):
+
+    table_theme = Parameter()
+
+    def requires(self):
+        return DownloadFRIncomeIris(table_theme=self.table_theme)
+
+    def xls2csv(self):
+        # Read in excel file, it should be the only file in self.input().path
+        # xls = pd.ExcelFile(os.path.join(self.input().path, os.listdir(self.input().path)[0]))
+        xls = pd.ExcelFile(glob.glob(os.path.join(self.input().path,
+                                                  '{table_theme}.xls'.format(table_theme=self.table_theme)))[0])
+
+        # Remove header
+        df = xls.parse(skiprows=5, header=0)
+        df.to_csv(os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme)),
+                  index=False, encoding='utf8')
+
+    def input_csv(self):
+        if not glob.glob(os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme))):
+            self.xls2csv()
+
+        return os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme))
+
+
+class SourceTags(TagsTask):
+    def version(self):
+        return 1
+
+    def tags(self):
+        return [
+            OBSTag(id='insee',
+                   name='INSEE',
+                   type='source',
+                   description='http://www.insee.fr/fr/bases-de-donnees/default.asp?page=recensement/resultats/2012/donnees-detaillees-recensement-2012.htm')
+        ]
+
+
+class IrisIncomeColumns(ColumnsTask):
+
+    table_theme = Parameter()
+
+    def requires(self):
+        return {
+            'sections': SectionTags(),
+            'subsections': SubsectionTags(),
+            'unittags': UnitTags(),
+            'sourcetag': SourceTags()
+        }
+
+
+    def version(self):
+        return 1
+
+    def columns(self):
+        cols = OrderedDict()
+        input_ = self.input()
+
+        subsectiontags = input_['subsections']
+        unittags = input_['unittags']
+        france = input_['sections']['fr']
+        insee_source = input_['sourcetag']['insee']
+
+        filepath = "incomemetadata/Income Variables - {}.tsv".format(self.table_theme)
+        with open(os.path.join(os.path.dirname(__file__),filepath)) as tsvfile:
+            tsvreader = csv.reader(tsvfile, delimiter="\t")
+            # Skip first row (header)
+            next(tsvreader, None)
+            for line in tsvreader:
+                var_code, short_name, long_name, universe, var_unit, \
+                    denominators, subsections = line
+                denominators = denominators.split(',')
+                targets_dict = {}
+                for x in denominators:
+                    x = x.strip()
+                    targets_dict[cols.get(x)] = 'denominator'
+                targets_dict.pop(None, None)
+                universes = universe.split(',')
+                for x in universes:
+                    x = x.strip()
+                    targets_dict[cols.get(x)] = 'universe'
```

Comment:
Becuase `universe` and `denominator` are mutually exclusive we should improve this part and make it more readable for the future